### ISSUE TITLE
rx.moment date prop should accept a `datetime`

### ISSUE THREAD

**Issue by masenf** [reflex-dev/reflex#4228]
**Describe the bug**
```python
class State(rx.State):
    date: datetime.datetime = datetime.datetime.now() - datetime.timedelta(days=1)

...

rx.moment(date=State.date)
```

```
TypeError: Invalid var passed for prop Moment.date, expected type <class 'str'>, got value reflex___state____state__repro_moment_locale___repro_moment_locale____state.date of type <class 'datetime.datetime'>.
```

**Specifics (please complete the following information):**
 - Python Version: 3.12
 - Reflex Version: `main`

**Comment by mistamun** [comment]
Same with form.input(type="date") I suppose

**Comment by masenf** [comment]
Reopening as this change was de-scoped from the merged PR

**Comment by picklelo** [comment]
Merged in #4381
